src/rule_message.cc

```diff
@@ -31,7 +31,7 @@ std::string RuleMessage::_details(const RuleMessage *rm) {
     msg.append(" [id \"" + std::to_string(rm->m_ruleId) + "\"]");
     msg.append(" [rev \"" + rm->m_rev + "\"]");
     msg.append(" [msg \"" + rm->m_message + "\"]");
-    msg.append(" [data \"" + utils::string::limitTo(200, rm->m_data) + "\"]");
+    msg.append(" [data \"" + utils::string::log_escape_hex(utils::string::limitTo(200, rm->m_data)) + "\"]");
```

Comment:
We probably shouldn't restrict encoding of the double-quote character to only the 'data' item. That character could plausibly also appear in other items that can be filled via variable expansion in a rule. The most obvious additional item is 'msg', but 'tag' could also be affected.

Reply:
Yes, you are right. I reviewed mod_security2's code, and found that more field are also escaped beside the `[data]` and as you mentioned the `[msg]`. These are also escaped:

[[id]](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/apache2/re.c#L2227-L2228), [[rev]](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/apache2/re.c#L2231-L2232), [[msg]](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/apache2/re.c#L2241-L2242), [[ver]](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/apache2/re.c#L2272-L2273), [[tag]](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/apache2/re.c#L2298-L2299)

I think that would be fine to add these too - then the v3 logformat would be even more compatible with v2.

What do you think?

Reply:
[msg] and [tag] for sure, per my earlier comments.

[ver] ==> this is probably unlikely, but, it does support variable expansion, so: yes, include that one too
[rev] ==> even more obscure since it doesn't support population from untrusted input, but since an admin could, in theory, hard-code the double-quote character, I guess I'll say: yes, do this one too
[id] ==> I'll say no here. It's a bit debatable, but since there's no way to include a double-quote character here, I'd lean towards avoiding an extra specific encoding call.

Reply:
This is done by commit 105c590.